### ISSUE TITLE
Migrate all @Entity annotations (columns and relations) from method level to field level

### ISSUE THREAD

**Issue by Brutus5000** [micheljung/faf-java-api#346]
The Elide library should support field-level hibernate annotations for quite a while now.

Since we use Lombok, creating the Getter-methods manually is unnecessary code and I would like to move the annotation to the field level variables.

In case new problems occur, we'd need to create a bug ticket at elide bug tracker (https://github.com/yahoo/elide/issues).

**Comment by vanogrid** [comment]
I will start working on this issue.

**Comment by Brutus5000** [comment]
Nice. I have added instructions to the readme how to setup our database, otherwise you can't run the test suite.

**Comment by 1-alex98** [comment]
Consider asking a moderator to join your chat service zulip @vanogrid

**Comment by nagavenkatakiran** [comment]
Hi Team,
I would like to contribute on this issue

**Comment by vanogrid** [comment]
Hi, @nagavenkatakiran 
Here is my status update: I moved the annotations to field level, but now there are 8 failing integration tests. I found out that the issue causing it are the lazy-fetched collections, for example: [here](https://github.com/FAForever/faf-java-api/blob/2c7853d57b6c99fcd8657df74b5fcc4a257be5f5/src/main/java/com/faforever/api/data/domain/UserNote.java#L36). When I set them to Eager fetching and replace the `repository.getOne(id)` calls in the code with `repository.findById(id)`, all tests pass. 

I think the problem is related to Elide using Hibernate Proxy objects, which don't have all fields hydrated. Unfortunately, I was able to set up a new Spring Boot project with Elide and lazy-fetched collections (no Spring Security, though) and the issue was not present, so I'm still not sure where the problem is exactly. I'm currently debugging Elide to see what it does when fetching those collections. There is a similar [issue](https://github.com/yahoo/elide/issues/844) in the Elide repo, but I will need to gather more info to see if it is the same issue or we need to file a new one. 

I hope this information will be of help to get you up to speed.
